Fails to build on Windows 10

Hello,
I've been struggling to build this package on ROS2 Eloquent on Windows 10.

All other packages (except those dependent on this) build fine, and running the command:
`colcon build --merge-install --packages-select resource_retriever --event-handler console_cohesion+`

Outputs
```
Starting >>> resource_retriever
--- output: resource_retriever
[6.968s] -- Found ament_cmake_ros: 0.8.0
(C:/dev/ros2_eloquent/install/share/ament_cmake_ros/cmake)
[7.375s] -- Using PYTHON_EXECUTABLE: C:/Python37/python.exe
[8.640s] -- Found ament_index_cpp: 0.7.2` `(C:/dev/ros2_eloquent/install/share/ament_index_cpp/cmake)
[8.703s] -- Found libcurl_vendor: 2.2.1` `(C:/dev/ros2_eloquent/install/share/libcurl_vendor/cmake)
[8.796s] -- library:
C:/dev/ros2_eloquent/install/opt/libcurl_vendor/bin/libcurl.dll
[9.234s] -- Configuring done
[9.828s] -- Generating done
[9.875s] -- Build files have been written to:` `C:/dev/ros2_eloquent/build/resource_retriever
[10.234s] Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Framework
[10.234s] Copyright (C) Microsoft Corporation. All rights reserved.
[10.234s]
[11.125s]` `C:\dev\ros2_eloquent\install\opt\libcurl_vendor\bin\libcurl.dll : fatal error LNK1107: invalid or corrupt file: cannot read at 0x310` `[C:\dev\ros2_eloquent\build\resource_retriever\resource_retriever.vcxproj]
---
Failed   <<< resource_retriever [ Exited with code 1 ]
 
Summary: 0 packages finished [12.6s]
1 package failed: resource_retriever
```

I've been looking at it and found that the source might come from
`# TODO(wjwwood): remove libcurl_vendor and just use system curl when possible`
`find_package(libcurl_vendor REQUIRED)`

In the CMakeLists.txt file, but I have no idea how to start and solve it.

--- a/libcurl_vendor/CMakeLists.txt
+++ b/libcurl_vendor/CMakeLists.txt
@@ -85,6 +85,7 @@ if(NOT CURL_FOUND)
     set(ENV_VAR_VALUE "opt/libcurl_vendor/lib")
   endif()
   ament_environment_hooks(env_hook/libcurl_vendor_library_path.dsv.in)
+  set(CURL_FOUND FALSE)
 endif()
 
 ament_package(

--- a/libcurl_vendor/libcurl_vendor-extras.cmake.in
+++ b/libcurl_vendor/libcurl_vendor-extras.cmake.in
@@ -1,3 +1,6 @@
+# Ignore the broken curl-config.cmake in 7.58
+set(CURL_NO_CURL_CMAKE ON)
+
 # Look for system curl first.
 find_package(CURL QUIET)
 

--- a/libcurl_vendor/libcurl_vendor-extras.cmake.in
+++ b/libcurl_vendor/libcurl_vendor-extras.cmake.in
@@ -1,5 +1,7 @@
 # Ignore the broken curl-config.cmake in 7.58
-set(CURL_NO_CURL_CMAKE ON)
+if(NOT @CURL_FOUND@)
+  set(CURL_NO_CURL_CMAKE ON)
+endif()
 
 # Look for system curl first.
 find_package(CURL QUIET)

